src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -834,6 +1109,101 @@ private void AddMetaKeyValues(int i, DataViewSchema.Annotations.Builder builder)
+                var srcShape = ctx.RetrieveShapeOrNull(srcVariable);
+                var srcType = _srcTypes[iinfo].GetItemType().RawType;
+                if (_srcTypes[iinfo] is KeyDataViewType) return false;
+
+                // Numeric input types are limited to those supported by the Onnxruntime MurmurHash operator, which currently only supports
+                // uints and ints. Thus, ulongs, longs, doubles, floats, and booleans are not supported.
+                if (srcType == typeof(ushort) || srcType == typeof(short) || srcType == typeof(sbyte) ||
+                    srcType == typeof(byte) || srcType == typeof(bool))
+                {
+                    castOutput = ctx.AddIntermediateVariable(NumberDataViewType.UInt32, "CastOutput", true);
+                    castNode = ctx.CreateNode("Cast", srcVariable, castOutput, ctx.GetNodeName(opType), "");
+                    castNode.AddAttribute("to", NumberDataViewType.UInt32.RawType);
+                    murmurNode = ctx.CreateNode(opType, castOutput, murmurOutput, ctx.GetNodeName(opType), "com.microsoft");
+                }
+                else if (srcType == typeof(uint) || srcType == typeof(int) || srcType == typeof(ReadOnlyMemory<char>))
+                {
+                    murmurNode = ctx.CreateNode(opType, srcVariable, murmurOutput, ctx.GetNodeName(opType), "com.microsoft");
+                }
+                else
+                {
+                    return false;
+                }
+
+                murmurNode.AddAttribute("positive", 1);
+                var seed = _parent._columns[iinfo].Seed;
+                if (_parent._columns[iinfo].UseOrderedHashing)
+                {
+                    if (_srcTypes[iinfo] is VectorDataViewType)
+                        return false;
+                    seed = Hashing.MurmurRound(seed, 0);
+                }
+                murmurNode.AddAttribute("seed", seed);
+
+                // masking is done via bitshifts, until bitwise AND is supported by Onnxruntime
+                opType = "BitShift";
+                string bitShiftOutput = ctx.AddIntermediateVariable(_dstTypes[iinfo], "BitShiftOutput");
+                var shiftValue = ctx.AddInitializer((ulong)(32 - _parent._columns[iinfo].NumberOfBits), false);
+                var shiftNode = ctx.CreateNode(opType, new[] { murmurOutput, shiftValue }, new[] { bitShiftOutput }, ctx.GetNodeName(opType), "");
+                shiftNode.AddAttribute("direction", "LEFT");
+
+                opType = "BitShift";
+                string bitShiftOutput2 = ctx.AddIntermediateVariable(_dstTypes[iinfo], "BitShiftOutput2");
+                shiftNode = ctx.CreateNode(opType, new[] { bitShiftOutput, shiftValue }, new[] { bitShiftOutput2 }, ctx.GetNodeName(opType), "");
+                shiftNode.AddAttribute("direction", "RIGHT");
+
+                var vectorShape = new VectorDataViewType(NumberDataViewType.Int64, (int)srcShape[1]);
```

Comment:
>srcShape [](start = 88, length = 8)

This can change to `_srcTypes[iinfo].VectorSize` as well (or `ValueCount`? Or is it `GetValueCount()`? I don't remember the exact syntax). Then you don't need `srcShape` at all.